api/src/main/java/io/minio/MinioClient.java

```diff
@@ -970,7 +971,7 @@ private Request createRequest(
       }
     }
 
-    if (contentEncoding != null) {
+    if (!Strings.isNullOrEmpty(contentEncoding)) {
```

Comment:
you can also filter out standard headers with empty value in `normalizeHeaders()`